docs/guide/fragments.md

```diff
@@ -1,61 +1,91 @@
+The only macros feature currently unavailable or unreplaced for fragments is passing the template context using `with context`. This is planned for fragments as well.
+:::
 
 ## Using fragments
 
-With that out of the way, how do they work?
-
-Fragments are defined with the `{% fragment %}` template tag, followed by a name for the fragment. The fragment closes with the `{% endfragment %}` tag.
+We define fragments by putting template markup between `{% fragment %}` and `{% endfragment %}` tags. The opening tag should also include a name for the fragment.
 
 ```django
 {% fragment heading() %}
   <h2>This is a heading fragment</h2>
 {% endfragment %}
 ```
 
-One difference from macros is that they are then called using the `render` template tag:
+To use the fragment in a template, reference it by name using the `{% render %}` tag:
 
 ```django
 {% render heading() %}
 {# Renders: `<h2>This is a heading fragment</h2>` #}
 ```
 
-You can also pass arguments to fragments, which is key to using them across templates. Those arguments can be any data or variables understood by Apostrophe templates (e.g., strings, objects, arrays, etc.).
+We can also pass arguments to fragments, allowing us to reuse them across templates with different data.
 
 ```django
 {% fragment heading(adjective) %}
   <h2>This is a {{ adjective }} fragment</h2>
 {% endfragment %}
 
+{# 👇 Passing in a string argument #}
 {% render heading('cool') %}
-{# Renders: `<h2>This is a cool fragment</h2>` #}
 ```
 
-## Importing fragments
+Fragment arguments can be any data or template variables understood by Apostrophe templates (e.g., strings, objects, arrays, etc.). It can often be useful to pass a [doc object](/reference/glossary.md#doc) to a macro to render its areas, especially when docs are referenced in [relationships](/guide/relationships.md).
 
-As shown above, fragments can be defined and used in a single template file. That can be useful to break up complicated markup, but it is often better to put the fragment into its own file for reuse or to improve mantainability. Like any template file, fragments are placed in a module's `views` directory, or the root-level `views` directory in an Apostrophe project.
+```django
+{% fragment authorCredit(author) %}
+  <div>
+    <p>By {{ author.title }}</p>
+    {% area author, 'photo' %}
+  </div>
+{% endfragment %}
 
-### Importing within a single module
+{# 👇 Passing in a piece object from a relationship field #}
+{% render authorCredit(data.piece._author) %}
+```
 
-For example, you might have an article [index page](/reference/glossary.md#index-page) that will list a series of linked article titles with rich text teasers. The index page template itself would be at the path `modules/article-page/views/index.html`. You could separate the markup for each item in the listing into a fragment file, `modules/article-page/views/item-fragment.html`.
+<!-- ::: tip
```

Comment:
I thought I included a note about this. That's because skipping positional arguments is buggy at the moment. This one https://github.com/apostrophecms/apostrophe/issues/3056